composite/proof-of-reserves/src/adapter.ts

```diff
@@ -1,56 +1,42 @@
 import { logger } from '@chainlink/external-adapter'
-import { AdapterRequest, AdapterResponse, Execute } from '@chainlink/types'
-import reduceAdapter from '@chainlink/reduce-adapter'
-import { getImpl as getProtocolImpl, getProtocol } from './protocol'
-import { getImpl as getBalanceImpl, getBitcoinIndexer } from './balance'
+import { AdapterRequest, ExecuteWithConfig, Config, ExecuteFactory } from '@chainlink/types'
+import { Validator, Requester } from '@chainlink/external-adapter'
+import { makeConfig, makeOptions, getURL } from './config'
+import { runProtocolAdapter } from './protocol'
+import { runBalanceAdapter } from './balance'
+import { runReduceAdapter } from './reduce'
+
+export const makeRequestFactory = (config: Config, prefix: string) => async (
+  input: AdapterRequest,
+) => Requester.request({ ...config.api, method: 'post', url: getURL(prefix, true), data: input })
 
 // Run, log, throw on error
-const runAdapter = async (execute: Execute, input: AdapterRequest, tag: string) => {
+export const callAdapter = async (execute: any, input: AdapterRequest, tag: string) => {
   const output = await execute(input)
   logger.debug(tag, { output })
   return output
 }
 
-// Get address set for protocol
-const runProtocolAdapter = async (input: AdapterRequest) => {
-  const execute = getProtocolImpl({ type: getProtocol() })
-  return runAdapter(execute, input, '_onProtocol')
+const inputParams = {
+  protocol: true,
+  indexer: true,
```

Comment:
Keeping this composite adapter very flexible. There are 2 protocol adapters and 7 balance adapters. Instead of running 14 different PoR adapters we can just run 1.

The two main concepts are: 
- input params to switch between underlying adapters
- `DATA_PROVIDER_URL` uses the param prefix to find the URL to the location

Reply:
It will not be possible to split the adapters based on input params, as the Chainlink node will have to send the exact same params to all adapters. Last time this issue came up I proposed we split this on query string params on the bridge instead, which would make it possible to deploy a single adapter, and split the protocol/balance adapters based on the QS params you set when creating the bridge. Maybe this is something we should start prioritizing instead?

Reply:
After our talk this looks good. Going to copy this to the other composite adapters as well 👍